dashboard/app/controllers/activities_controller.rb

```diff
@@ -24,7 +24,7 @@ def milestone
 
     if params[:script_level_id]
       @script_level = ScriptLevel.cache_find(params[:script_level_id].to_i)
-      @level = @script_level.level
+      @level = (params[:level_id] && Level.find(params[:level_id].to_i)) || @script_level.levels[0]
```

Comment:
nit: a ternary operator would probably be clearer here, but that's arguably a matter of personal preference.